compiler/aarch64/codegen/BinaryEvaluator.cpp

```diff
@@ -615,6 +615,63 @@ OMR::ARM64::TreeEvaluator::vsubEvaluator(TR::Node *node, TR::CodeGenerator *cg)
    return inlineVectorBinaryOp(node, cg, subOp);
    }
 
+TR::Register *
+OMR::ARM64::TreeEvaluator::vmulInt64Helper(TR::Node *node, TR::CodeGenerator *cg)
+   {
+   /*
+    * Vector mul instruction on aarch64 does not support 64-bit interger element.
+    *
+    * Suppose we have two long integers (64 bits), AB and CD, where each of A, B, C, and D is one word (32 bits).
+    * To get their product, AB * CD, we can do the following:
+    *
+    *  AB * CD = B*D + 2^32*(A*D + B*C) + 2^64*(A*C)
+    *
+    *  Since we are only keeping the lower 64 bits of the result, this is equivalent to:
+    *
+    *  AB * CD = B*D + 2^32*(A*D + B*C)
+    *
+    *  To perform (A*D + B*C), we reverse elements in AB and do 32-bit element wise multiplication for BA and CD.
+    *
+    */
+   TR::Node *firstChild = node->getFirstChild();
+   TR::Node *secondChild = node->getSecondChild();
+   TR::Register *lhsReg = cg->evaluate(firstChild);
+   TR::Register *rhsReg = cg->evaluate(secondChild);
+
+   TR::Register *tmp1Reg = cg->allocateRegister(TR_VRF);
+   TR::Register *tmp2Reg = cg->allocateRegister(TR_VRF);
+   TR::Register *productReg = cg->allocateRegister(TR_VRF);
+
+   TR_ARM64ScratchRegisterManager *srm = cg->generateScratchRegisterManager();
+
+   /* Reverse 32bit words in 64bit elements */
+   generateTrg1Src1Instruction(cg, TR::InstOpCode::vrev64_4s, node, tmp1Reg, lhsReg);
+
+   generateTrg1Src2Instruction(cg, TR::InstOpCode::vmul4s, node, tmp2Reg, tmp1Reg, rhsReg);
+   /*
+    * At this moment, upper 32bit of each 64-bit element of tmp2Reg is A*D, lower 32bit is B*C.
+    * Get A*D + B*C into the lower 32bit.
+    */
+   generateVectorShiftImmediateInstruction(cg, TR::InstOpCode::vushr2d, node,tmp1Reg, tmp2Reg, 32);
```

Comment:
Please add a space between `node,` and `tmp1Reg`.

Reply:
fixed.